js/views/userPage/TypeFilter.js

```diff
@@ -0,0 +1,91 @@
+/* categoryFilter.js and typeFilter.js share quite a bit. Ensure that when one is updated,
+the other is also maintained.
+*/
+
+import $ from 'jquery';
+import _ from 'underscore';
+import loadTemplate from '../../utils/loadTemplate';
+import BaseVw from '../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    const opts = {
+      ...options,
+    };
+
+    opts.initialState = {
+      types: [],
+      selected: 'all',
+      expanded: false,
+      maxInitiallyVisibleTypes: 6,
+      ...(options.initialState || {}),
+    };
+
+    super(opts);
+    this.options = opts;
+    this._state = opts.initialState;
+  }
+
+  className() {
+    return 'clrP clrBr padMd clrT contentBox clrSh2 form veryCompact categoryOrTypeFilter';
+  }
+
+  events() {
+    return {
+      'click .js-showMoreLess': 'onClickShowMoreLess',
+      'change input[type="radio"]': 'onChangeType',
+    };
+  }
+
+  onClickShowMoreLess() {
+    this.setState({ expanded: !this.getState().expanded });
+  }
+
+  onChangeType(e) {
+    this._state.selected = e.target.value;
```

Comment:
You're setting state.selected here, but retrieving this._selected in get selectedType. I don't think selectedType is used anywhere?

Reply:
Yeah, that's a copied over oversight from `CategoryFilter.js`. We could probably also remove `get selectedCat() {` from that view.

Reply:
Another thing we've changed going forward is generally we don't modify the `_state` object directly. We do it via `setState`. Up to you if you want to update things in that regard. (I know you were just using a file as a template that was probably written a year and a half ago)